src/allotropy/parsers/utils/values.py

```diff
@@ -31,6 +33,15 @@ def natural_sort_key(key: str) -> list[str]:
 T = TypeVar("T")
 
 
+def assert_df(
```

Comment:
Couldn't this functionality be achieved using `assert_not_none`?

Reply:
`assert_df` was created to make the type checker happy: without it, I get complaints in `src/allotropy/parsers/perkin_elmer_envision/perkin_elmer_envision_structure.py` because it isn't sure that the object in hand is not `None`, so it doesn't want me to call dataframe methods.

Reply:
ok, I'll take a look at this in follow-up.